hax/hax/motr/hax.c

```diff
@@ -251,11 +251,16 @@ PyObject *m0_ha_proc_counters_fetch(unsigned long long ctx,
 		PyList_SET_ITEM(list, i, pver_bc);
 	}
 
+	/* Allocation of pc_bckey and pc_bcrec happens in
+	* m0_spiel_proc_counters_fetch in motr code */
+	if(len != 0)
+		m0_spiel_count_stats_free(&count_stats, len);
```

Comment:
Better, but it would still be better there's corresponding init/alloc function for count_stats. Allocating part of the structure silently and then expecting user to invoke a corresponding free call is not very good.
I would suggest to separate out initialization of conut_stats structure and have a corresponding fini call.
m0_spiel_count_stats_init(), m0_spiel_proc_counters_fetch() and m0_spiel_count_stats_fini()